### PR TITLE
[FLINK-15623][checkstyle] Exclude target directory

### PR DESCRIPTION
**Pull request by zentol**
Excludes the entire `target` directory from checkstyle. 

We only care about checkstyle for our own source that we actively maintain, not generated sources or other stuff in this directory.

### CODE REVIEW

File: tools/maven/suppressions.xml [hunk 1]
```diff
@@ -32,13 +32,6 @@ under the License.
 		<suppress files="org[\\/]apache[\\/]beam[\\/]vendor[\\/]grpc[\\/]v1p21p0[\\/]io[\\/]netty[\\/]buffer.*.java" checks="[a-zA-Z0-9]*"/>
 		<suppress files="org[\\/]apache[\\/]beam[\\/]runners[\\/]fnexecution[\\/]environment.*.java" checks="[a-zA-Z0-9]*"/>
 
-		<!-- Python streaming API follows python naming conventions -->
-		<suppress
-			files="org[\\/]apache[\\/]flink[\\/]streaming[\\/]python[\\/]api[\\/].*.java"
-			checks="MethodNameCheck|ParameterName"/>
-		<suppress
-			files="org[\\/]apache[\\/]flink[\\/]streaming[\\/]python[\\/]connectors[\\/].*.java"
-		checks="MethodNameCheck|ParameterName"/>
```
Comment (tillrohrmann) [inline tools/maven/suppressions.xml]: Why can we remove these suppressions? According to the comment, the Java Python files follow python naming conventions?

Comment (tillrohrmann) [reply tools/maven/suppressions.xml]: Is it because these files are only in the `target` directory?

Comment (zentol) [reply tools/maven/suppressions.xml]: those are part of the python streaming api we removed a long time ago.

Comment (tillrohrmann) [reply tools/maven/suppressions.xml]: Thanks for the clarification.

**Pull request merged**